### ISSUE TITLE
Option to fail on warning

### ISSUE THREAD

**Issue by dnault** [asciidoctor/asciidoctor-gradle-plugin#154]
I'd like to fail the build when AsciiDoctor issues a warning. To reproduce, process a document with an include directive for a non-existent file, like:

include::doesNotExist.txt[]

**Comment by rhaendel** [comment]
+1

**Comment by aalmiray** [comment]
@mojavelinux @lordofthejars : we would need asciidoctor to bubble up the waring through asciidoctorj so that the plugin can configure it. Does asciidoctor (ruby) have a message/logger handler already? perhaps we just need to expose it through asciidoctorj

**Comment by ysb33r** [comment]
We should actually have this as an option in asciidoctor itself, so that asciidoctorj (and such the plugin) can just set it. The failure will then be handled automatically in the same way that it is handled by the plugin today.

**Comment by mojavelinux** [comment]
This is already planned in Asciidoctor core. We shouldn't try to patch a solution until we get it solved there. Then it becomes a simple matter of reading the message stack and failing if it contains violations. See https://github.com/asciidoctor/asciidoctor/issues/44.

**Comment by mojavelinux** [comment]
This issue should be marked as "tracking". I like to have to label in the issue tracker for this reason.

**Comment by otrosien** [comment]
I have an intermediate workaround for this problem: Add a stdout listener on the asciidoctor task, and check for "WARNING", or in my case, for missing include files. Not something to be put into this plugin, actually, but I'll add it here so people can bridge the gap until this feature is supported.

```
asciidoctor {
            ext.capturedOutput = [ ]
            def listener = { ext.capturedOutput << it } as StandardOutputListener

            logging.addStandardErrorListener(listener)
            logging.addStandardOutputListener(listener)

            doLast {
                logging.removeStandardOutputListener(listener)
                logging.removeStandardErrorListener(listener)
                ext.capturedOutput.join('').with { output ->
                    if (output =~ /include file not found:/) {
                        throw new RuntimeException("Include file(s) not found.\n" + output)
                    }
                }
            }
}
```

**Comment by mojavelinux** [comment]
Very good. This is exactly how this should be solved in the interim. I agree that putting it into the plugin or in AsciidoctorJ is too risky. But putting it in the build configuration is both safe and transparent.

Perhaps we can add this to one of the examples in asciidoctor-gradle-examples.

**Comment by otrosien** [comment]
courtesy of Kimble at stackoverflow, btw. http://stackoverflow.com/questions/31523166/gradle-capturing-output-written-to-out-err-on-a-per-task-basis

There seems to be an issue with parallel task execution mentioned there, which I haven't tried to reproduce, as I'm not executing asciidoctor tasks in parallel.

**Comment by vogella** [comment]
For reference, @fap- solved this for our build via https://github.com/vogellacompany/com.vogella.gradle.plugins/commit/7a286f7b714ddb177085f957b63b07a8698676ca

**Comment by ysb33r** [comment]
Now that we have the LogHandler in AsciidoctorJ (1.5.7, 1.6.0-alpha.7) we can probably deal with this. I'll look to see if I can add this to 2.0.x development.

**Comment by ysb33r** [comment]
The code for this will be merged from PR #251 and be available in 2.0-alpha.3

To cover the exact case of this issue the global or task exteniosn can be configured as
 
```groovy
asciidoctorj {
   fatalWarnings missingIncludes()
}
```

More generically any pattern can be supplied, for instance

```groovy
asciidoctorj {
   fatalWarnings ~/include file not found|missing callout/
}
```

**Comment by adamu** [comment]
Kotlin:

```kotlin
tasks {
    asciidoctor {
        asciidoctorj.fatalWarnings(listOf(asciidoctorj.missingIncludes()))
    }
}
```
